### ISSUE TITLE
Online snapshot cannot be restored when CPU hotplug is enabled

### ISSUE THREAD

**Issue by kreeuwijk** [kubevirt/kubevirt#10481]
**What happened**:
When I create a VM snapshot of a running VM, the snapshot contains a running configuration of the VM. If CPU hotplug is enabled (through `liveUpdateFeatures` in the VM spec), the snapshot fails to restore with the following message on the `vmrestore` object:

```
status:
  complete: false
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2023-09-26T10:35:27Z"
    reason: 'admission webhook "virtualmachine-validator.kubevirt.io" denied the request:
      CPU topology maxSockets cannot be set directy in VM template'
    status: "False"
    type: Progressing
```

This is caused by the `maxSockets` property existing on the `spec.domain.cpu` section when the VM is running.
As there is no way to resolve the situation, the VM remains in the `restoreInProgress` state forever and can no longer be started. The only resolution is to delete the VM and start over.

If the VM is shutdown first and then an offline snapshot is taken, that snapshot can be restored successfully.

**What you expected to happen**:
Online snapshots should be restorable even when CPU hotplug is enabled on the VM. The snapshot procedure should not capture the `.spec.domain.cpu.maxSockets` property on the running VM, or alternatively, this property should not be restored to the VM spec.

Also, there should be a way to abort a snapshot restore operation that is encountering problems.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a VM with CPU hotplug enabled
```
spec:
  liveUpdateFeatures:
    cpu: {}
```
2. Start the VM and wait for it to be fully running
3. Snapshot the VM while it is running
4. Shutdown the VM
5. Restore the online snapshot
6. Notice the `vmrestore` object goes to a `COMPLETE` state of `false` immediately
7. Describe the `vmrestore` object to see the status
```
admission webhook "virtualmachine-validator.kubevirt.io" denied the request: CPU topology maxSockets cannot be set directy in VM template
```

**Environment**:
- KubeVirt version (use `virtctl version`): 1.0.0
- Kubernetes version (use `kubectl version`): 1.27.2
- VM or VMI specifications: CPU hotplug enabled
- Cloud provider or hardware configuration: bare metal (Canonical MAAS)
- OS (e.g. from /etc/os-release): Ubuntu
- Kernel (e.g. `uname -a`): 5.15.0-79-generic
- Install tools: N/A
- Others: N/A

**Comment by enp0s3** [comment]
Hey @kreeuwijk!

Thank you for the reporting of the issue!

I've created a PR to fix the issue, but ATM we don't officially support hotplug with combination of other functionalities (the feature is still under a feature-gate, thus not stable). In this particular case we are missing a proper coverage in tests for this scenario, and there is a trade-off when adding such tests, because it will require more CI resources (current hotplug implementation involves live-migration, in the future we plan to integrate the in-place pod update, which should be less CI consuming)

I am working on graduation plan for the hotplug, as part of a dynamic VM reconfiguration subject. I will let you know as soon as it will be published.

**Comment by acardace** [comment]
@enp0s3 could this be fixed with https://github.com/kubevirt/kubevirt/pull/10515?

**Comment by fabiand** [comment]
@kreeuwijk thanks for providing this feedback.

Did you see #11054 which is changing the API of this feature?

**Comment by kreeuwijk** [comment]
That's interesting. Will it however change the culprit of this problem, which is that the CRD does not allow setting `.spec.domain.cpu.maxSockets` at all?

**Comment by xpivarc** [comment]
> That's interesting. Will it however change the culprit of this problem, which is that the CRD does not allow setting `.spec.domain.cpu.maxSockets` at all?

I would encourage to retest but the answer should be yes https://github.com/kubevirt/kubevirt/pull/11054/files#diff-c4943461fdddd6e77c8bcd7824fa0448402d00efaf473a55c4604f17e99d7461L451-L455

**Comment by kreeuwijk** [comment]
@xpivarc I can retest to confirm; which Kubevirt release is this merged in?

**Comment by xpivarc** [comment]
> @xpivarc I can retest to confirm; which Kubevirt release is this merged in?

This is going to be included in 1.2, you should be able to try it out with https://github.com/kubevirt/kubevirt/releases/tag/v1.2.0-beta.0
